packages/calendar-time-utils/src/formats.ts

```diff
@@ -0,0 +1,69 @@
+// Reference: https://momentjs.com/docs/#/displaying/format/
+export enum FormatToken {
+  D = 'D', // NUMERIC_SINGLE_DAY
+  DD = 'DD', // NUMERIC_TWO_DIGITS_DAY
+  M = 'M', // NUMERIC_SINGLE_MONTH
+  MM = 'MM', // NUMERIC_TWO_DIGIT_MONTH
+  YY = 'YY', // TWO_DIGITS_YEAR
+  YYYY = 'YYYY', // FOUR_DIGITS_YEAR
+  h = 'h', // HOUR_12_SINGLE
+  hh = 'hh', // HOUR_12_TWO_DIGITS
+  H = 'H', // HOUR_24_SINGLE
+  HH = 'HH', // HOUR_24_TWO_DIGITS
+  m = 'm', // MINUTE_SINGLE
+  mm = 'mm', // MINUTE_TWO_DIGITS
+  A = 'A', // PERIOD - https://unicode.org/reports/tr35/tr35-6.html#Date_Format_Patterns
+}
```

Comment:
Here we define an enumeration with all the possible characters moment locale pattern might include so we can then _translate_ them to the expected value based on the locale.